src/Fcf/Classes.hs

```diff
@@ -13,7 +13,24 @@ module Fcf.Classes
 
 import Fcf.Core
 
+-- $
+-- >>> import Fcf.Core
+-- >>> import Fcf.Utils
+-- >>> import Fcf.Data.Nat
+-- >>> import qualified GHC.TypeLits as TL
+
+
 -- | Type-level 'fmap' for type-level functors.
+--
+-- === __Example__
+--
+-- >>> data AddMul :: Nat -> Nat -> Exp Nat
+-- >>> type instance Eval (AddMul x y) = x TL.* x TL.+ 2 TL.* x TL.* y TL.+ y TL.* y
```

Comment:
The factored form would be shorter and easier to read.

Reply:
ok changed